shell/edit/networking.k8s.io.ingress/IngressClass.vue

```diff
@@ -37,8 +37,7 @@ export default {
   },
   methods: {
     update(e) {
-      this.ingressClassName = e && typeof e === 'object' ? e.value : e;
-
+      this.ingressClassName = e.value !== undefined ? e.value : e.label;
```

Comment:
You're right about needing to check that e is defined, but I had to go back and run this code to remember why I used `undefined` specifically here, which seems like a good indication that it is not clear enough. So I've refactored the update method. BUT if you're curious as to why I used `e.value !== undefined`, it was because of the way data is emitted from labeled select when the user defines their own option. In that case, the event emitted looks like:
```
{
   label: <user value>
}
``` 

When the user selects 'None', the event emitted looks like

```
{
   label: 'None',
   value: '' 
}
```

So if we did `e && !!e.value ? e.value : e.label` we'd set the ingressClassName to "None" when that option is selected instead of deleting that key.